src/sharepoint/rest/lists.ts

```diff
@@ -41,7 +41,7 @@ export class Lists extends QueryableCollection {
      */
     public getById(id: string): List {
         let list = new List(this);
-        list.concat(`(guid'${id}')`);
+        list.concat(`(guid'${id.replace(/[^a-zA-Z0-9_-]/g,"")}')`);
```

Comment:
Does it make sense to limit the regex to only supported characters (a-f)? Or to replace the "{" & "}" directly?